Automattic/wp-calypso
# Use the wc/v3/continents endpoint to get the list of countries and states for the Shipping Labels UI

I recommend reviewing this PR commit by commit. It's big-ish but the commits are small and focused.

To test:
* Go purchase a shipping label.
* See that all the countries with a USPS post office appear in the Origin Address country field.
* Select `United States`.
* See that every state appears in the `State` field, including the military bases.
* Select `Puerto Rico`.
* Check that the `State` field is now a text field.
* Validate the address.
* See that the summary string of the step displays the correct country and state names.
* Go to the `Destination` address.
* See that all the countries are listed in the `Country` dropdown.
* Put a valid European or Canadian address and validate it.
* Go to the `Customs` step.
* See that you can choose any country in the world for the `Origin country` field.
* Go to the `config/development.json` file, remove the `woocommerce/extension-wcservices/international-labels` line, restart Calypso.
* Repeart the previous steps, check that the only available countries for both `Origin` and `Destination` address are `US`, `Puerto Rico` and `Virgin Islands`, check that you can't see the military bases when you choose `US`.
* Check that you can't get to the `Purchase Label` modal if the order destination is in a place that requires a Customs form (outside of `US`, `Puerto Rico` or `Virgin Islands`).
